### PR TITLE
Add moment-matching support to `$loo()` method

### PR DESCRIPTION
**Pull request by andrjohns**
#### Submission Checklist

- [x] Run unit tests
- [x] Declare copyright holder and agree to license (see below)

#### Summary

This PR adds an option to the `$loo()` method: `moment_match`, which enables the use of a moment-matching correction for any problematic observations: https://mc-stan.org/loo/articles/loo2-moment-matching.html

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting
(this will be you or your assignee, such as a university or company):
Andrew Johnson

By submitting this pull request, the copyright holder is agreeing to
license the submitted work under the following licenses:

- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

**Comment by topipa** [comment]
Thank you for implemeting this! Let me know if I can help with anything. The implementation looks good to me, although I'm not very familiar with `cmdstanr`.

**Comment by andrjohns** [comment]
> Thank you for implemeting this! Let me know if I can help with anything. The implementation looks good to me, although I'm not very familiar with `cmdstanr`.

No worries! Feel free to let me know if there are any other additions/features that could be useful

**Review by rok-cesnovar — APPROVED** [review]
Looks good to me. Feel free to merge.

File: tests/testthat/resources/stan/loo_moment_match
Comment (rok-cesnovar) [inline tests/testthat/resources/stan/loo_moment_match:1]: This binary file should be removed.

**Comment by jgabry** [comment]
Cool, thanks for implementing this!

**Pull request merged**